### PR TITLE
kelly/header

### PR DESCRIPTION
**Pull request by kellybinary**
update language dropdown and header.

### CODE REVIEW

**Comment by ashkanx** [comment]
@kellybinary please fix the conflicts

File: templates/partials/language.mustache [hunk 1]
```diff
@@ -1,9 +1,9 @@
-	<span class="language-select">
+<!-- <span class="language-select">
```
Comment (aminmarashi) [inline templates/partials/language.mustache]: Never keep the previous code in comments, git will take care of that.

File: templates/bot.mustache [hunk 2]
```diff
@@ -1,168 +1,164 @@
...
-                      <div class="logo">
-                          <img class="responsive" src="https://style.binary.com/images/logo/symbol.svg" alt="Binary-logo" />
-                      </div>
-                      <div class="binary-logo-text">
-                          <img class="responsive" src="https://style.binary.com/images/logo/type.svg" alt="Binary-logo">
-                      </div>
-                  </div>
-              </a>
-          </div>
+  <head>
```
Comment (aminmarashi) [inline templates/bot.mustache]: @kellybinary It's always a good idea to pull the most recent changes from upstream  before you make a PR. Because sometimes changes from other collaborators will cause merge conflicts.
👍 for fixing the indentation.

File: templates/bot.mustache [hunk 3]
```diff
@@ -1,168 +1,164 @@
...
             <tr>
-              <th><span data-i18n-text="Number"></span></th>
-              <th><span data-i18n-text="Reference"></span></th>
-              <th><span data-i18n-text="Trade Type"></span></th>
-              <th><span data-i18n-text="Entry Spot"></span></th>
-              <th><span data-i18n-text="Exit Spot"></span></th>
-              <th><span data-i18n-text="Buy Price"></span></th>
-              <th><span data-i18n-text="Final Price"></span></th>
-              <th><span data-i18n-text="Profit/Loss"></span></th>
+              <th data-i18n-text="No. of runs"></th>
```
Comment (aminmarashi) [inline templates/bot.mustache]: 👍 for fixing the other parts of the code, better to make a separate commit for this fix.

File: static/css/main.css [hunk 4]
```diff
@@ -11,13 +11,19 @@ html, body {
     border-top: 2px solid #394171;
     border-bottom: 4px solid #e98024;
 }
-#header .wrapper {
+#header .wrapper, #topbar .wrapper{
```
Comment (aminmarashi) [inline static/css/main.css]: `.wrapper` selector already exists.

File: src/common/translator.js [hunk 5]
```diff
@@ -17,73 +17,93 @@ import en from './translations/en/i10n.json'
...
-  vi,
-  pl,
-  ru,
-  pt,
-  es,
-  fr,
-  en,
-  ach,
-}
+    zh_tw: zhTw,
```
Comment (aminmarashi) [inline src/common/translator.js]: @kellybinary We use 2 spaces for each tab, not 4.

File: templates/bot.mustache (hunk 3 above)
Comment (kellybinary) [inline templates/bot.mustache]: i didn't add this tho.. it was already there, probably fetched from the beta branch

File: src/common/translator.js (hunk 5 above)
Comment (kellybinary) [inline src/common/translator.js]: noted. i will fix it.

File: src/common/language_dropdown.js [hunk 6]
```diff
@@ -0,0 +1,14 @@
+import $ from 'jquery'
+
+export default class LanguageDropdown {
```
Comment (aminmarashi) [inline src/common/language_dropdown.js]: :1st_place_medal: for using ES6.
This function can be moved inside the translator constructor and be used instead of the event listener like this:
```
$('ul#select_language li')
  .click(function onClick() {
     const lang = $(this).attr('class')
     document.location.search = `l=${lang}`
  })
....
```

Comment (aminmarashi) [inline src/common/language_dropdown.js]: In es6 encapsulation is the default behavior, therefore you don't need to define classes to ensure encapsulation. You could simply write:
```
export () => {
  $('ul#select_language li')
    .click(function onClick() {
...
}

```

File: src/common/translator.js [hunk 7]
```diff
@@ -30,14 +30,34 @@ const supportedLanguages = {
...
+    $('.language').on('change_language', (event, value) => {
+      document.location.search = `l=${value}`
+    })
+    $.each(languageDisplayName, (key, value) => { // populate language dropdown
+      $('#select_language').append(`<li class=${key}>${value}</li>`);
     })
     const lang = this.getLanguage()
-    $('#language').val(lang)
+    $('.language').text(languageDisplayName[lang]);
+    $(document.getElementsByClassName(lang)[0]).hide();
```
Comment (aminmarashi) [inline src/common/translator.js]: shorter version thanks to es6:
```
$(`.${lang}`)
  .hide()
```

File: src/common/translator.js [hunk 8]
```diff
@@ -30,14 +30,34 @@ const supportedLanguages = {
...
+
 export default class Translator {
   constructor() {
-    $('#language').change(e => {
-      document.location.search = `l=${e.target.value}`
+    $('.language').on('change_language', (event, value) => {
+      document.location.search = `l=${value}`
+    })
+    $.each(languageDisplayName, (key, value) => { // populate language dropdown
+      $('#select_language').append(`<li class=${key}>${value}</li>`);
```
Comment (aminmarashi) [inline src/common/translator.js]: 👍

File: src/common/translator.js [hunk 9]
```diff
@@ -30,14 +30,34 @@ const supportedLanguages = {
...
+};
+
 export default class Translator {
   constructor() {
-    $('#language').change(e => {
-      document.location.search = `l=${e.target.value}`
+    $('.language').on('change_language', (event, value) => {
+      document.location.search = `l=${value}`
+    })
+    $.each(languageDisplayName, (key, value) => { // populate language dropdown
```
Comment (aminmarashi) [inline src/common/translator.js]: Better not to use jquery each sth js is able to do:
```
for (const key of Object.keys(languageDisplayName)) {
  const value = languageDisplayName[key]
...
```

**Comment by aminmarashi** [comment]
@ashkanx Should we do the same for the ugly tour dropdown?
screen shot 2016-12-27 at 11 17 49 am

**Comment by ashkanx** [comment]
Yes. I think it's better to make it look like the other one.

File: src/common/translator.js (hunk 7 above)
Comment (kellybinary) [inline src/common/translator.js]: lovely shorter version!

File: src/common/translator.js (hunk 9 above)
Comment (kellybinary) [inline src/common/translator.js]: noted +1

File: src/common/language_dropdown.js (hunk 6 above)
Comment (kellybinary) [inline src/common/language_dropdown.js]: :smile: you're right, needed to wrap it in another click event as such:
`    $('.language')
      .click(function onClick() {
        $('ul#select_language li')
          .click(function onClick() {
            const lang = $(this).attr('class')
            document.location.search = `l=${lang}`
          })
      })`
